lib/portage/sync/syncbase.py

```diff
@@ -252,6 +252,12 @@ def _refresh_keys(self, openpgp_env):
 		"""
 		out = portage.output.EOutput(quiet=('--quiet' in self.options['emerge_config'].opts))
 
+		# Pass proxy settings in emerge configuration to Gemato ephemeral env
+		running_config_settings = self.options['emerge_config'].running_config.settings
+		for setting in ('http_proxy', 'https_proxy'):
+			if running_config_settings.get(setting, None):
+				os.environ[setting] = running_config_settings.get(setting)
```

Comment:
I've been thinking about it. If we don't need a version with this stable very fast, I'll add a `proxy=` parameter to key refresh functions. This will be better for when I replace GPG with a pure Python implementation.

Reply:
I would vote at first to have a non-blocking case like currently with this PR but that depends also on you and your time for doing your modifications. In both cases, thank you for advancing the subject.